postgres-appliance/configure_spilo.py

```diff
@@ -210,16 +211,21 @@ def deep_update(a, b):
 
 def get_provider():
     try:
-        logging.info("Figuring out my environment (Google? AWS? Local?)")
+        logging.info("Figuring out my environment (Google? AWS? Openstack? Local?)")
         r = requests.get('http://169.254.169.254', timeout=2)
         if r.headers.get('Metadata-Flavor', '') == 'Google':
             return PROVIDER_GOOGLE
         else:
-            r = requests.get('http://169.254.169.254/latest/meta-data/ami-id')  # should be only accessible on AWS
-            if r.ok:
-                return PROVIDER_AWS
-            else:
-                return PROVIDER_UNSUPPORTED
+            try:
+                r = requests.get('http://instance-data/latest/meta-data/ami-id')  # accessible on AWS, will fail on Openstack
+                if r.ok:
+                    return PROVIDER_AWS
+            except (requests.exceptions.ConnectTimeout, requests.exceptions.ConnectionError):
```

Comment:
According to the https://github.com/zalando/spilo/issues/183 it will raise `socket.gaierror: [Errno -2] Name or service not known` (fail to resolve instance-data) on openshift and therefore will to fix the problem

Reply:
@CyberDem0n the problem was actually discovered in plain Kuberenetes running on OpenStack. However, If a user was running OpenShift on OpenStack, they would likely encounter the same error.

Reply:
I understand, but before your changes if was failing in a different place, because `socket.gaierror:` exception hasn't been handled properly.
Now it will fail on the line 220, because DNS can't resolve `instance-data` into ip address.
On the line 223 we are trying to catch only connection error exceptions.
If you do a change:
```python
 -            except (requests.exceptions.ConnectTimeout, requests.exceptions.ConnectionError): 
 +            except Exception: 
```

It will start working, but in addition to that I think it would make sense to change line 224:
```python
-                r = requests.get('http://169.254.169.254/latest/meta-data/ami-id')
+                r = requests.get('http://169.254.169.254/openstack/latest/meta_data.json')
```

Reply:
@CyberDem0n I have implemented the changes you suggested, I removed instance-data and replaced it with the metadata server ip address which should resolve #191